askama_parser/src/node.rs

```diff
@@ -24,7 +24,7 @@ pub enum Node<'a> {
     LetDecl(Ws, Target<'a>),
     Let(Ws, Target<'a>, Expr<'a>),
     Cond(Vec<Cond<'a>>, Ws),
-    Match(Ws, Expr<'a>, Vec<When<'a>>, Ws),
+    Match(Ws, Expr<'a>, Vec<Node<'a>>, Vec<When<'a>>, Ws),
```

Comment:
I think we should express in the type system that we only parse comments here, not just any node. Probably create a separate type `struct Comment<'a> { ws: Ws, text: &'a str }`, change `Comment` to contain that (this should happen as a separate commit before this one), and then change the added type here to be `Vec<Comment<'a>>`.

Reply:
:100: and this was a source of much confusion for me when working on this rebase, reminding myself that only one node type was valid there.  I'll implement your suggestion.

Reply:
I've elected to leave this out for now, plan to add it back in a future PR.